src/Domain.cpp

```diff
@@ -703,6 +704,15 @@ void Domain::enableComponentwiseThermostat()
 	}
 }
 
+void Domain::setComponentThermostat(int cid, int thermostat) {
+	if ((0 > cid) || (0 >= thermostat)) {
+		Log::global_log->error() << "Domain::setComponentThermostat: cid or thermostat id too low" << std::endl;
+		mardyn_exit(787);
+	}
+	this->_componentToThermostatIdMap[cid] = thermostat;
+	this->_universalThermostatN[thermostat] = 0;
+}
+
```

Comment:
did you mean to put this into this PR?

Reply:
Nope thats good! Implementations belong into `cpp` files (except for templated functions).